### ISSUE TITLE
Defining a method with 4 required arguments

### ISSUE THREAD

**Issue by p-mongo** [jruby/jruby#6244]
https://github.com/jruby/jruby/wiki/JRubyMethod_Signatures says:

> Specific argument counts are supported for 0 to 3 arguments. All signatures that require more than 3 arguments must use an argument array.

The wiki page does not provide any argument array examples that I see though.

I tried:

<pre>
    @JRubyMethod(name = "initialize")
    public IRubyObject initialize(ThreadContext context, IRubyObject[] args) throws Throwable {}
</pre>

This produced:

<pre>
java.lang.RuntimeException: IRubyObject[] args but neither of optional or rest specified for method org.mongodb.ruby.gssapi.Authenticator.initialize
</pre>

Full output: https://gist.github.com/p-mongo/e181d235b6332464e92c8c06412d5b23

How do I define this method with 4 required arguments please?

**Comment by headius** [comment]
Yeah unfortunately we don't support 4 explicit arguments. It's a mechanical limitation we could fix with a bunch of effort, but have not because 1-3 args covers 99% of all functions.

You will want to provide both an `IRubyObject[]` argument and a `required = 4` in the `@JRubyMethod` annotation for the number of required arguments. If there are optional arguments, use `optional = #` and if there's no upper limit, use `rest = true`.
